src/arizona_live_view.erl

```diff
@@ -26,38 +26,31 @@ Live view.
 %% API functions.
 -export([parse_str/2]).
 -ignore_xref([parse_str/2]).
--export([compile/3]).
--ignore_xref([compile/3]).
--export([persist_get/3]).
 -export([mount/2]).
 -export([handle_event/4]).
-
 -export([put_macro/3]).
 -ignore_xref([put_macro/3]).
 -export([get_macro/3]).
 -ignore_xref([get_macro/3]).
 
--opaque macros() :: map().
+-type macros() :: map().
 -export_type([macros/0]).
 
--opaque tree() :: list().
--export_type([tree/0]).
-
-%% Macros
--define(PERSIST_KEY, ?MODULE).
-
 %% --------------------------------------------------------------------
 %% Callbacks.
 %% --------------------------------------------------------------------
 
--callback mount(arizona_socket:t()) ->
-    {ok, arizona_socket:t()}.
+-callback mount(Socket) -> Socket
+    when Socket :: arizona_socket:t().
 
--callback render(macros()) ->
-    tree().
+-callback render(Macros) -> Tree
+    when Macros :: macros(),
+         Tree :: arizona_tpl_parse:tree().
 
--callback handle_event(EventName :: binary(), arizona:payload(), arizona_socket:t()) ->
-    {noreply, arizona_socket:t()}.
+-callback handle_event(EventName, Payload, Socket) -> Socket
+    when EventName :: binary(),
+         Payload :: arizona:payload(),
+         Socket :: arizona_socket:t().
```

Comment:
Same here about same input and output:
```suggestion
-callback handle_event(EventName, Payload, Socket0) -> Socket1
    when EventName :: binary(),
         Payload :: arizona:payload(),
         Socket0 :: arizona_socket:t(),
         Socket1 :: arizona_socket:t().
```

Reply:
Same as above. Unless you want to rename it for reasons like clarity, but it should be pretty obvious that the `Socket` changed, in any case (why else would we return it?).

Reply:
I'm okay with keeping it as is, no problem.

Reply:
a83a068